implants/lib/eldritch/src/pivot/ssh_exec_impl.rs

```diff
@@ -1,5 +1,205 @@
 use anyhow::Result;
+use starlark::{values::{dict::Dict, Heap, Value}, collections::SmallMap, const_frozen_string};
 
-pub fn ssh_exec(_target: String, _port: i32, _username: String, _password: String, _key: String, _command: String, _shell_path: String) -> Result<String> {
-    unimplemented!("Method unimplemented")
+use super::Session;
+
+struct SSHExecOutput {
+    stdout: String,
+    status: i32,
+}
+
+async fn handle_ssh_exec(target: String, port: u16, command: String, username: String, password: Option<String>, key: Option<String>, key_password: Option<&str>, timeout: Option<u32>) -> Result<SSHExecOutput> {
+    let mut ssh = tokio::time::timeout(
+        std::time::Duration::from_secs(timeout.unwrap_or(3).into()),
+        Session::connect(username, password, key, key_password, format!("{}:{}", target, port)),
+
+    ).await??;
```

Comment:
the rare double `?`

Reply:
Errors are caught in the parent function `ssh_exec`